### PR TITLE
Splunk On-Call plugin: REST endpoint

### LINKED ISSUE #4555: Splunk On-Call plugin: REST endpoint
As mentioned by @adamdmharvey [here](https://github.com/backstage/backstage/pull/4416#discussion_r573373010), we should use the Splunk On-Call recommended way (REST endpoint) to create, resolve and acknowledge incidents.

## Feature Suggestion

Implement the Splunk On-Call REST Endpoint.

## Possible Implementation

See the [REST Endpoint integration guide](https://help.victorops.com/knowledge-base/rest-endpoint-integration-guide/) for more information.

For the moment the plugin uses the api which is documented [here](https://portal.victorops.com/api-docs/).

Here are some of the points we need to do to implement this feature:

- Add the routing key and the integration url in the entity annotations:

    ```yaml
    annotations:
      splunk.com/integration-url': <SPLUNK_ON_CALL_URL_TO_NOTIFY>
      splunk.com/routing-key': <SPLUNK_ON_CALL_ROUTING_KEY>
    ```
- Updates the `acknowledgeIncident`, `resolveIncident` and `triggerAlarm` methods inside `plugins/splunk-on-call/src/api/client.ts` to use the integration url.
- Updates the README with the new guidelines to be followed.
- Update the `TriggerDialog` component accordingly ([glossary of fields table](https://help.victorops.com/knowledge-base/incident-fields-glossary/#glossary-of-fields))

## Context

PR which added Splunk On-Call plugin - #4416

### PR DESCRIPTION
**Pull request by ayshiff**
## Hey, I just made a Pull Request!

Closes #4555

This PR introduce the Splunk On-Call recommended way (REST endpoint) to create, resolve and acknowledge incidents.

The incident creation now requires the following information:

- Incident type (`CRITICAL`, `WARNING`, `INFO`)
Note that only `CRITICAL` incidents will be displayed on the incident list (like on the Splunk On-Call dashboard).   
The others incidents will be displayed in the Timeline section of the portal.
<p align="center">

</p>

- Incident start time (*not required*)
- Incident id (*not required*)
- Incident display name
- Incident message

<p align="center">

</p>

Since the creation of incidents now goes through the REST endpoint, we no longer have to provide a username in the annotations.
Incidents that were created with the REST Endpoint appear with the creator name `{ REST }`.

I also added some things:

- A link to the Splunk On-Call portal
- The incident number in the incident list like in the Splunk On-Call portal.
- A scroll in the incident and OnCall lists (in case we have a lot of elements)

<p align="center">

</p>

For some reason the changes applied to incidents (trigger-resolve-acknowledge) may take some time to actually be applied after receiving the response from the server.
This is the reason why I added a timeout before the incident GET request otherwise we had cases where an incident could be correctly created and the response of the GET `/incidents` request did not contain it. (same for resolve and acknowledge)

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [X] Tests for new functionality and regression tests for bug fixes
- [X] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4719-09rmgurrlgwzrvamxexdorf53gfaygzl.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6038fe91f285afea569984b3

File: plugins/splunk-on-call/src/components/Escalation/EscalationPolicy.tsx [hunk 1]
```diff
@@ -24,12 +30,28 @@ import { useApi, Progress } from '@backstage/core';
...
 import { User } from '../types';
 
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      maxHeight: '400px',
+      overflow: 'auto',
+    },
+    subheader: {
+      backgroundColor: 'white',
```
Comment (Rugvip) [inline plugins/splunk-on-call/src/components/Escalation/EscalationPolicy.tsx:40]: Best use a color from the theme palette here, or dark mode is gonna look strange

File: plugins/splunk-on-call/src/components/Incident/Incidents.tsx [hunk 2]
```diff
@@ -15,24 +15,50 @@
...
 import { Alert } from '@material-ui/lab';
 
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      maxHeight: '400px',
+      overflow: 'auto',
+    },
+    subheader: {
+      backgroundColor: 'white',
```
Comment (Rugvip) [inline plugins/splunk-on-call/src/components/Incident/Incidents.tsx:39]: same here

File: plugins/splunk-on-call/src/components/Incident/Incidents.tsx [hunk 3]
```diff
@@ -15,24 +15,50 @@
...
 export const Incidents = ({ refreshIncidents, team }: Props) => {
+  const classes = useStyles();
   const api = useApi(splunkOnCallApiRef);
 
   const [{ value: incidents, loading, error }, getIncidents] = useAsyncFn(
     async () => {
+      // For some reason the changes applied to incidents (trigger-resolve-acknowledge)
+      // may take some time to actually be applied after receiving the response from the server.
+      // The timeout compensates for this latency.
+      await new Promise(resolve => setTimeout(resolve, 2000));
```
Comment (Rugvip) [inline plugins/splunk-on-call/src/components/Incident/Incidents.tsx:61]: Oh no 😅 , don't have a better suggestion that doesn't complicate things a lot more tho :p

**Review by Rugvip — COMMENTED** [review]
Thanks! Thinking it's ready for review :>

File: plugins/splunk-on-call/src/components/Incident/Incidents.tsx (hunk 3 above)
Comment (ayshiff) [reply plugins/splunk-on-call/src/components/Incident/Incidents.tsx:61]: This method doesn't really suit me 😅  
Maybe we could use polling to check every `n-time` if we have the new status of the incidents.

I am open to other suggestions 👍

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4719-09rmgurrlgwzrvamxexdorf53gfaygzl.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6038fe91f285afea569984b3

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4719-09rmgurrlgwzrvamxexdorf53gfaygzl.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6038fe91f285afea569984b3

File: .changeset/rude-roses-jam.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/plugin-splunk-on-call': minor
+---
+
+Updated splunk-on-call plugin to use the REST endpoint (incident creation-acknowledgement-resolution).
+It implies switching from `splunkOnCall.username` annotation to `splunkOnCall.eventsRestEndpoint` annotation, this is a breaking change.
```
Comment (Rugvip) [inline .changeset/rude-roses-jam.md:6]: annotation -> configuration

**Comment by backstage-service** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr4719-09rmgurrlgwzrvamxexdorf53gfaygzl.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6038fe91f285afea569984b3

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**